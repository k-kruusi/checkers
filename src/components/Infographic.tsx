import { useMemo } from 'react';
import { ReactComponent as TimeIcon } from '../assets/icons/timer.svg';
import { ReactComponent as PiecesIcon } from '../assets/icons/pieces.svg';
import { theme } from '../theme';
import { Piece } from '../schema';
import { useCheckers } from '../hooks';
import { getPieces, getString, isTurn } from '../utils';

const outterContainer: React.CSSProperties = {
  border: `4px solid ${theme.colors.gold}`,
  borderRadius: 5,
  width: 'fit-content',
  minWidth: 85,
  userSelect: 'none',
  fontFamily: theme.fonts.amatic,
};

const rowStyle: React.CSSProperties = {
  display: 'flex',
  flexDirection: 'row',
  alignItems: 'center',
  gap: 5,
  textAlign: 'left',
  fontSize: 18,
};

const headerStyle: React.CSSProperties = {
  fontSize: 18,
  textAlign: 'center'
};

// renders the player data
export const InfoGraphic = ({ piece }: {
  piece: Piece
}) => {

  const { state, blackTime, redTime } = useCheckers();
  const { board, turn } = state;
  const { red, black } = getPieces(board);
  const isBlackTurn = isTurn(turn.phase, Piece.Black);
  const isMyTurn = isBlackTurn && piece === Piece.Black || !isBlackTurn && piece === Piece.Red;
  const myTitle = piece === Piece.Black ? getString('black') : getString('red');
  const myTime = piece === Piece.Black ? blackTime : redTime;
  const myCount = piece === Piece.Black ? black.length : red.length;

  const innerContainer: React.CSSProperties = useMemo(() => {
    return {
      margin: 5,
      padding: '5px 10px',
      backgroundColor: isMyTurn ? theme.colors.ivory : theme.colors.grey,
      fontWeight: 'bold',
      textAlign: 'left',
    };
  }, [isMyTurn]);

  return (<div style={outterContainer}>
    <div style={innerContainer}>
      <div><h1 style={headerStyle}>{myTitle}</h1></div>
      <div style={rowStyle}><TimeIcon /> <p>{myTime}</p></div>
      <div style={rowStyle}><PiecesIcon /> <p>{myCount}</p></div>
    </div>
  </div>);
};