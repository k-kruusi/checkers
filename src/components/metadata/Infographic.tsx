
import { theme } from "../../theme";
import { ReactComponent as TimeIcon } from '../../assets/icons/timer.svg';
import { ReactComponent as PiecesIcon } from '../../assets/icons/pieces.svg';
import { useMemo } from "react";

const outterContainer: React.CSSProperties = {
  border: `4px solid ${theme.colors.gold}`,
  borderRadius: 5,
  width: 'fit-content',
  minWidth: 85,
  userSelect: 'none',
  fontFamily: theme.fonts.amatic,
};

const rowStyle: React.CSSProperties = {
  display: 'flex',
  flexDirection: 'row',
  alignItems: 'center',
  gap: 5,
  textAlign: 'left',
  fontSize: 18,
};

const headerStyle: React.CSSProperties = {
  fontSize: 18,
  textAlign: 'center'
};

export const InfoGraphic = ({ name, time, count, myTurn }: {
  name: string,
  time: string,
  count: number,
  myTurn: boolean
}) => {

  const innerContainer: React.CSSProperties = useMemo(() => {
    return {
      margin: 5,
      padding: "5px 10px",
      backgroundColor: myTurn ? theme.colors.ivory : theme.colors.grey,
      fontWeight: 'bold',
      textAlign: 'left',
    };
  }, [myTurn]);

  return (<div style={outterContainer}>
    <div style={innerContainer}>
      <div><h1 style={headerStyle}>{name}</h1></div>
      <div style={rowStyle}><TimeIcon /> <p>{time}</p></div>
      <div style={rowStyle}><PiecesIcon /> <p>{count}</p></div>
    </div>
  </div>);
};