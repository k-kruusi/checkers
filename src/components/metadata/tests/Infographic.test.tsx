import { render } from '@testing-library/react';
import { InfoGraphic } from '../Infographic';

test('InfoGraphic component snapshot', () => {
  const { asFragment } = render(
    <InfoGraphic name="Test Name" time="10:00" count={5} myTurn={true} />
  );
  expect(asFragment()).toMatchSnapshot();
});