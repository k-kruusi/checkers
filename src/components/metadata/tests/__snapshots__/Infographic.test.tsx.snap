// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoGraphic component snapshot 1`] = `
<DocumentFragment>
  <div
    style="border: 4px solid #daa520; border-radius: 5px; min-width: 85px; user-select: none; font-family: Amatic SC, sans-serif;"
  >
    <div
      style="margin: 5px; padding: 5px 10px; background-color: rgb(255, 255, 240); font-weight: bold; text-align: left;"
    >
      <div>
        <h1
          style="font-size: 18px; text-align: center;"
        >
          Test Name
        </h1>
      </div>
      <div
        style="display: flex; flex-direction: row; align-items: center; gap: 5px; text-align: left; font-size: 18px;"
      >
        <svg>
          timer.svg
        </svg>
         
        <p>
          10:00
        </p>
      </div>
      <div
        style="display: flex; flex-direction: row; align-items: center; gap: 5px; text-align: left; font-size: 18px;"
      >
        <svg>
          pieces.svg
        </svg>
         
        <p>
          5
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
